CLAUDE.md

```diff
@@ -0,0 +1,285 @@
+# CLAUDE.md
+
+This file provides guidance to Claude Code (claude.ai/code) when working with code in this repository.
+
+## lib-jitsi-meet Architecture
+
+This is the JavaScript library for accessing Jitsi Meet server-side deployments. It provides WebRTC functionality, XMPP communication, and media handling for Jitsi Meet clients.
+
+## Common Development Commands
+
+### Build Commands
+```bash
+npm run build           # Full build (webpack UMD bundle + TypeScript compilation)
+npm run build:webpack   # Build UMD bundle only for browser <script> tags
+npm run build:webpack-dev # Development webpack bundle
+npm run build:tsc       # TypeScript compilation only for ESM modules
+npm run watch           # Development build with file watching
+```
+
+### Development Commands
+
+**Testing:**
+- `npm test` - Run all tests via Karma (single run)
+- `npm run test-watch` - Run tests in watch mode
+- Tests use Jasmine framework with Chrome headless browser
+
+**Code Quality:**
+- `npm run lint` - ESLint + TypeScript type checking
+- `npm run lint-fix` - Auto-fix linting issues
+- `npm run type-check` - TypeScript type checking only
+
+**Documentation:**
+- `npm run typedoc` - Generate TypeScript documentation
+
+**Other:**
+ - `npm run gen-types` - Generate TypeScript declaration file
+
+### TypeScript Migration
+
+New features should be implemented only with TypeScript. The codebase is actively migrating from JavaScript to TypeScript:
+
+**Migration Strategy:**
```

Comment:
This section adds nothing, I'd remove it.

Reply:
Agree